src/vivarium/framework/results/context.py

```diff
@@ -95,12 +100,58 @@ def add_stratification(
         stratification = Stratification(name, sources, categories, mapper, is_vectorized)
         self.stratifications.append(stratification)
 
+    def register_stratified_observation(
```

Comment:
We have discussed at length whether the interface should have a different registration function per observation type or a single `register_observation` function that takes a "type" arg. That's a bit tricky to implement but might be cleaner? idk - I'm open to thoughts